core/src/test/java/cucumber/runtime/formatter/PrettyFormatterTest.java

```diff
@@ -381,6 +382,35 @@ public void should_mark_arguments_in_steps() throws Throwable {
                                           AnsiEscapes.GREEN + AnsiEscapes.INTENSITY_BOLD + "arg2"  + AnsiEscapes.RESET));
     }
 
+    @Test
+    public void should_mark_nested_argument_as_part_of_full_argument(){
+        Formats formats = new AnsiFormats();
+        Argument enclosingArg = new Argument(20, "and not yet confirmed");
+        Argument nestedArg = new Argument(-17, "not yet ");
+        PrettyFormatter prettyFormatter = new PrettyFormatter(null);
+
+        String formattedText = prettyFormatter.formatStepText("Given ", "the order is placed and not yet confirmed", formats.get("passed"), formats.get("passed_arg"), asList(enclosingArg, nestedArg));
+
+        assertThat(formattedText, equalTo(AnsiEscapes.GREEN + "Given " + AnsiEscapes.RESET +
+            AnsiEscapes.GREEN + "the order is placed " + AnsiEscapes.RESET +
+            AnsiEscapes.GREEN + AnsiEscapes.INTENSITY_BOLD + "and not yet confirmed"  + AnsiEscapes.RESET));
+    }
+
+    @Test
+    public void should_mark_nested_arguments_as_part_of_enclosing_argument(){
+        Formats formats = new AnsiFormats();
+        Argument enclosingArg = new Argument(20, "and not yet confirmed");
+        Argument nestedArg = new Argument(-17, "not yet ");
+        Argument nestedNestedArg = new Argument(-5, "yet ");
```

Comment:
I had forgotten to check if the examples were correct by writing an integration test. I have done so now. 

In JdkPatternArgumentMatcherTest you can add this test to reproduce the examples:

```
    @Test
    public void canHandleNestedCaptureGroups() throws UnsupportedEncodingException {
        assertVariables("the order is placed( and( not yet)? confirmed)?", "the order is placed and not yet confirmed",
            " and not yet confirmed", 19,
            " not yet", 23);
    }
```

Important to note is that the indexes are not negative. They just come before the end of the previous capture group.